python/lsst/cp/pipe/utilsEfd.py

```diff
@@ -257,41 +315,163 @@ def parseMonochromatorStatus(self, data, dateStr):
-        # in question is found.
-        low = 0
-        high = len(data)
-        idx = (high + low) // 2
-        found = False
-        iteration = 0
-        while not found:
-            if idx < 0 or idx > len(data) or iteration > 10:
-                raise RuntimeError("Search for date failed?")
+        Parameters
+        ----------
+        dataSeries : `str`, optional
+            Data series to request from the EFD.
+        dateMin : `str`, optional
+            Minimum date to retrieve from EFD.
+        dateMax : `str`, optional
+            Maximum date to retrieve from EFD.
 
-            myTime = data["private_sndStamp"][idx]
+        Returns
+        -------
+        results : `astropy.table.Table`
+            The table of results returned from the EFD.
+        """
+        # All electrometer data gets written to the same series.
+        defaultSeries = "lsst.sal.Electrometer.logevent_intensity"
+        alternateSeries = "lsst.sal.Electrometer.logevent_logMessage"
+        dataSeries = dataSeries if dataSeries else defaultSeries
 
-            if myTime <= dateValue:
-                low = idx
-            elif myTime > dateValue:
-                high = idx
+        if dateMin:
+            startDate = Time(dateMin, format='isot', scale='tai')
+        else:
+            startDate = None
+        if dateMax:
+            stopDate = Time(dateMax, format='isot', scale='tai')
+        else:
+            stopDate = None
 
-            idx = (high + low) // 2
-            iteration += 1
-            if high - low == 1:
-                found = True
-            self.log.debug("parse search %d %d %d %d %s %s",
-                           low, high, idx, found, myTime, dateValue)
+        if (dataSeries == alternateSeries and (startDate is None or stopDate is None)):
+            raise RuntimeError("Cannot query logevent_logMessage without dates to limit memory issues.")
 
-        # End binary search.
+        results = self.selectTimeSeries(dataSeries, [],
+                                        startDate, stopDate)
+        if dataSeries == 'lsst.sal.Electrometer.logevent_logMessage':
+            results = self.rewriteElectrometerStatus(results)
+        return results
+
+    def rewriteElectrometerStatus(self, inResults):
+        """Rewrite intermediate electrometer data from the EFD.
+
+        Parameters
+        ----------
+        inResults : `astropy.table.Table`
+            The table of results returned from the EFD.
+
+        Returns
+        -------
+        outResults : `astropy.table.Table`
+            The rewritten table containing only electrometer summary
+            status events.
+        """
+        # This is fragile against upstream changes.
+        # Ignore all entries that are not the ones we care about.
+        outResults = inResults[np.where(inResults['functionName'] == 'write_fits_file')]
+        outResults = outResults[np.where(outResults['level'] == 20)]
```

Comment:
The `np.where` is superfluous here.  You can just do `inResults[inResults["functionName"] == "write_fits_fit"]`, etc.